core/queryresultio/sparqljson/src/main/java/org/eclipse/rdf4j/query/resultio/sparqljson/AbstractSPARQLJSONParser.java

```diff
@@ -342,14 +342,16 @@ else if (jp.getCurrentName().equals(DISTINCT) || jp.getCurrentName().equals(ORDE
 		} catch (JsonProcessingException e) {
 			throw new QueryResultParseException("Could not parse SPARQL/JSON", e, e.getLocation().getLineNr(),
 					e.getLocation().getLineNr());
+		} finally {
+			jp.close();
```

Comment:
maybe it's wrong to close jp, since the input stream comes from the user

Reply:
Could be subject of another ticket, since changing that to not closing the jp seems to be backwards incompatible ?